fission/src/ui/panels/configuring/MatchModeConfigPanel.tsx

```diff
@@ -0,0 +1,272 @@
+            </Box>
+        </Box>
+    )
+}
+
+const MatchModeConfigPanel: React.FC<PanelPropsImpl> = ({ panelId }) => {
+    const { closePanel } = usePanelControlContext()
+    const { openModal } = useModalControlContext()
+
+    const [matchModeConfigs, setMatchModeConfigs] = useState<MatchModeConfig[]>([])
+
+    useEffect(() => {
+        const loadAllJsonFiles = async () => {
+            try {
+                const indexRes = await fetch("match-mode-config/index.json")
+                const fileNames: string[] = await indexRes.json()
+
+                const defaultConfigs = await Promise.all(
+                    fileNames.map(async fileName => {
+                        const res = await fetch(`match-mode-config/${fileName}`)
+                        return res.json()
+                    })
+                )
+                const localConfigs = JSON.parse(window.localStorage.getItem("match-mode-configs") || "[]")
+
+                const combinedConfigs = [...defaultConfigs, ...localConfigs]
+                const uniqueConfigsById = Array.from(new Map(combinedConfigs.map(item => [item.id, item])).values())
+
+                setMatchModeConfigs(uniqueConfigsById)
+            } catch (err) {
+                console.error("Error loading JSON files:", err)
+                Global_AddToast?.(
+                    "error",
+                    "Error Loading Match Mode Configs",
+                    "Please check the console for more details"
+                )
+            }
+        }
+
+        loadAllJsonFiles()
+    }, [])
+
+    const matchModeConfigElements = useMemo(
+        () =>
+            matchModeConfigs.map(config =>
+                ItemCard({
+                    id: config.id,
+                    name: config.name || config.id || "Unnamed Match Mode",
+                    primaryOnClick: () => {
+                        MatchConfigSelected(config, openModal)
+                        closePanel("match-mode-config")
+                    },
+                    secondaryOnClick: () => {
+                        // Delete the config from the local storage
+                        setMatchModeConfigs(prev => prev.filter(c => c.id !== config.id))
+                        window.localStorage.setItem(
+                            "match-mode-configs",
+                            JSON.stringify(matchModeConfigs.filter(c => c.id !== config.id))
+                        )
+                        matchModeConfigs.filter(c => c.id !== config.id)
+                    },
+                })
+            ),
+        [matchModeConfigs, openModal, closePanel]
+    )
+
+    const fileUploadRef = useRef<HTMLInputElement>(null)
+
+    const uploadClicked = () => {
+        if (fileUploadRef.current) {
+            fileUploadRef.current.click()
+        }
+    }
+
+    // eslint-disable-next-line
+    const validateMatchModeConfig = (config: any): config is MatchModeConfig => {
```

Comment:
Because `MatchModeConfig` has all of these properties as required and not optional this type guard is not accurately representing whether `config` is an instance of `MatchModeConfig`. I think it makes most sense to deal with assigning the defaults in here rather than outside of the function